src/server/status_server.rs

```diff
@@ -614,17 +651,43 @@ mod tests {
     }
 
     #[test]
-    fn test_security_status_service() {
+    fn test_security_status_service_with_cn_validation() {
         let mut status_server = StatusServer::new(1, dummy_future_scheduler());
-        let _ = status_server.start("127.0.0.1:0".to_string(), &new_security_cfg());
-        let p = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+        let mut allowed_cn = HashSet::default();
+        allowed_cn.insert("tikv-server".to_owned());
+        let _ = status_server.start(
+            "127.0.0.1:0".to_string(),
+            &new_security_cfg(Some(allowed_cn)),
+        );
 
         let mut connector = HttpConnector::new(1);
         connector.enforce_http(false);
         let mut ssl = SslConnector::builder(SslMethod::tls()).unwrap();
+        ssl.set_certificate_file(
```

Comment:
Better keep the original case to show that without CN validation it's OK to connect without certificates.